cargo-shuttle/src/lib.rs

```diff
@@ -448,197 +449,212 @@ impl Shuttle {
-                println!("{log}");
+                err
+            })?;
+
+            let load_request = tonic::Request::new(LoadRequest {
+                path: executable_path
+                    .into_os_string()
+                    .into_string()
+                    .expect("to convert path to string"),
+                service_name: service_name.clone(),
+                resources: Default::default(),
+                secrets,
+            });
+            trace!("loading service");
+            let response = runtime_client
+                .load(load_request)
+                .or_else(|err| async {
+                    provisioner_server.abort();
+                    runtime.kill().await?;
+
+                    Err(err)
+                })
+                .await?
+                .into_inner();
+
+            if !response.success {
+                error!(error = response.message, "failed to load your service");
+                exit(1);
             }
-        });
 
-        println!("{resources}");
-
-        let addr = if run_args.external {
-            std::net::IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0))
-        } else {
-            Ipv4Addr::LOCALHOST.into()
-        };
+            let resources = response
+                .resources
+                .into_iter()
+                .map(resource::Response::from_bytes)
+                .collect();
+
+            let resources = get_resources_table(&resources, &service_name);
+
+            let mut stream = runtime_client
+                .subscribe_logs(tonic::Request::new(SubscribeLogsRequest {}))
+                .or_else(|err| async {
+                    provisioner_server.abort();
+                    runtime.kill().await?;
+
+                    Err(err)
+                })
+                .await?
+                .into_inner();
+
+            tokio::spawn(async move {
+                while let Ok(Some(log)) = stream.message().await {
+                    let log: shuttle_common::LogItem = log.try_into().expect("to convert log");
+                    println!("{log}");
+                }
+            });
 
-        let addr = SocketAddr::new(addr, run_args.port);
+            println!("{resources}");
 
-        let start_request = StartRequest {
-            ip: addr.to_string(),
-        };
+            let addr = if run_args.external {
+                std::net::IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0))
+            } else {
+                Ipv4Addr::LOCALHOST.into()
+            };
 
-        trace!(?start_request, "starting service");
-        let response = runtime_client
-            .start(tonic::Request::new(start_request))
-            .or_else(|err| async {
-                provisioner_server.abort();
-                runtime.kill().await?;
+            let addr = SocketAddr::new(
+                addr,
+                portpicker::pick_unused_port().expect("unable to find available port"),
```

Comment:
I opted for using portpicker here, but then I am ignoring the port arg the user passed in to start the service. But they can only pass in one port, and just using that +=1 for each service seems less reliable. I'll think about this one.

Reply:
Yip, once we have the router, that will again take care of only the one port the user wants to bind to. So +=1 will only be a temp thing if we go with that.